### PR TITLE
fs.mergeVolumes: Make a plan based on volumes size

### PR DESCRIPTION
**Pull request by yeganemehr**
# What problem are we solving?
I had 500 half-full volumes on my cluster, this means 15TB of wasted disk, It was hard and time-consuming to merge volumes individually.

# How are we solving the problem?
So i wrote a function to create a map (named `plan`) based on current status of volumes and decide how volumes must merge.
This function respect replication policy, volume TTL, collection, read-only volumes and max volumes size limit.

Now it's possible to
+ Merge volumes of a single collection
+ Merge only with specified destination volume (`-toVolumeId`)
+ Merge only with specified source volume (`-fromVolumeId`)

# How is the PR tested?
I run this locally and on my cluster.

### CODE REVIEW

File: weed/shell/command_fs_merge_volumes.go [hunk 1]
```diff
@@ -40,7 +44,7 @@ func (c *commandFsMergeVolumes) Help() string {
 	
 	This would help clear half-full volumes and let vacuum system to delete them later.
 
-	fs.mergeVolumes -toVolumeId=y [-fromVolumeId=x] [-apply] /dir/
+	fs.mergeVolumes [-toVolumeId=y] [-fromVolumeId=x] [-collection="*"] [-apply] [/dir/]
```
Comment (chrislusf) [inline weed/shell/command_fs_merge_volumes.go:47]: How about changing the directory into a named parameter also, in a separate PR?

File: weed/shell/command_fs_merge_volumes.go [hunk 2]
```diff
@@ -192,6 +195,94 @@ func (c *commandFsMergeVolumes) reloadVolumesInfo(masterClient *wdclient.MasterC
...
+}
+
+func (c *commandFsMergeVolumes) getVolumeSizeById(vid needle.VolumeId) uint64 {
+	return c.getVolumeSize(c.volumes[vid])
+}
+
+func (c *commandFsMergeVolumes) printPlan(plan map[needle.VolumeId]needle.VolumeId) {
+	fmt.Printf("max volume size: %d MB\n", c.volumeSizeLimit/1024/1024)
+	reversePlan := make(map[needle.VolumeId][]needle.VolumeId)
+	for src, dist := range plan {
```
Comment (chrislusf) [inline weed/shell/command_fs_merge_volumes.go]: dist => dest?

File: weed/shell/command_fs_merge_volumes.go (hunk 1 above)
Comment (yeganemehr) [reply weed/shell/command_fs_merge_volumes.go:47]: OK, after you merged this I'll send another PR. Is it okay with you?

File: weed/shell/command_fs_merge_volumes.go (hunk 2 above)
Comment (yeganemehr) [reply weed/shell/command_fs_merge_volumes.go]: done in 5091260

**Pull request merged**